scomplex/complex_image.go

```diff
@@ -29,33 +29,39 @@ func FromComplexArray(cpx []complex128, width int) (dst *ComplexImage) {
 	dst = new(ComplexImage)
 	dst.Pix = cpx
 	dst.Rect = image.Rect(0, 0, width, len(cpx)/width)
-	dst.MinRe = math.MaxFloat64
-	dst.MinIm = math.MaxFloat64
-	dst.MaxRe = -math.MaxFloat64
-	dst.MaxIm = -math.MaxFloat64
-	dst.MaxMod = 0.0
-	for _, c := range cpx {
+	dst.SetScaling()
+	return
+}
+
+func (comp *ComplexImage) SetScaling() {
+	comp.Rect = image.Rect(0, 0, width, len(cpx)/width)
```

Comment:
cpx is defined only for FromComplexArray and not passed in. The line is redundant as well, as the line remains in FromComplexArray. In ToShiftedComplex the Rect is taken directly from the Bounds of the src image, so it isn't needed. I suspect this line should be removed. In fact I'm surprised this compiles.